cms/static/js/views/course_video_settings.js

```diff
@@ -245,42 +268,85 @@ function($, Backbone, _, gettext, moment, HtmlUtils, StringUtils, TranscriptSett
             }
         },
 
-        renderLanguages: function() {
+        renderTargetLanguages: function() {
             var self = this,
+                availableLanguages,
                 $languagesPreferenceContainer = self.$el.find('.transcript-languages-wrapper'),
                 $languagesContainer = self.$el.find('.languages-container');
 
             // Clear error state if present any.
-            this.clearPreferenceErrorState($languagesPreferenceContainer);
+            self.clearPreferenceErrorState($languagesPreferenceContainer);
 
             $languagesContainer.empty();
 
-            // Show language container if provider is 3PlayMedia, else if fidelity is selected.
-            if (self.selectedProvider === THREE_PLAY_MEDIA || self.selectedFidelityPlan) {
-                self.availableLanguages = self.getPlanLanguages();
-                _.each(self.activeLanguages, function(activeLanguage) {
+            // Show language container if fidelity or source language is selected .
+            if (self.selectedVideoSourceLanguage || self.selectedFidelityPlan) {
+                availableLanguages = self.getTargetLanguages();
+                _.each(self.activeLanguages, function(language) {
                     // Only add if not in the list already.
-                    if (_.indexOf(self.selectedLanguages, activeLanguage) === -1) {
-                        self.selectedLanguages.push(activeLanguage);
-                        self.addLanguage(activeLanguage);
+                    if (_.indexOf(self.selectedLanguages, language) === -1) {
+                        self.selectedLanguages.push(language);
                     }
+                    // Show active/ add language language container
+                    self.addLanguage(language);
                 });
                 $languagesPreferenceContainer.show();
                 self.populateLanguageMenu();
             } else {
-                self.availableLanguages = {};
                 $languagesPreferenceContainer.hide();
             }
         },
 
+        renderVideoSourceLanguageMenu: function() {
+            var self = this,
+                availableLanguages = self.getTargetLanguages(),
+                availableTranslations = self.getSourceTranslations(),
+                $videoSourceLanguageContainer = self.$el.find('.video-source-language-wrapper'),
+                $languageMenuEl = self.$el.find('.video-source-language'),
+                selectOptionEl = new Option(gettext('Select language'), '');
+
+            if (availableTranslations) {
```

Comment:
Can we avoid this if condition and make decision when we call this method?

Reply:
Yes we can but it will introduce more complex code and the other method would have to do more work that this method should do